yb-voyager/src/query/queryparser/helpers_protomsg.go

```diff
@@ -370,3 +385,21 @@ func GetSchemaAndObjectName(nameList protoreflect.List) (string, string) {
 	}
 	return schemaName, objectName
 }
+
+func TraverseAndExtractDefNamesFromDefElem(msg protoreflect.Message) []string {
+	var defNames []string
+
+	collectorFunc := func(msg protoreflect.Message) error {
+		if GetMsgFullName(msg) != PG_QUERY_DEFELEM_NODE {
+			return nil
+		}
+
+		fieldVal := GetStringField(msg, "defname")
+		defNames = append(defNames, fieldVal)
+		return nil
+	}
+	visited := make(map[protoreflect.Message]bool)
+	TraverseParseTree(msg, visited, collectorFunc)
+
```

Comment:
You can also write a detector similar to FuncCallDetector, etc., for this. Instead of storing all the defNames, report unsupported def names, e.g., `security_invoker`, there itself?
This helps to keep the generic feature detection with Traverse in a single place.

Reply:
Discussed that with Aneesh, detection can't be just based on DefElem, since there are chances of same parameters like `security_invoker`, `on_error` etc being available with multiple DDLs.
So for making this into detector, we will also need to check objectType.
Refer - https://github.com/yugabyte/yb-voyager/pull/2108#:~:text=Regarding%20%2D%20Generic,security%20invoker)

So lets defer it for now, we can add if later we find it worth doing that.